### ISSUE TITLE
Rebalancer : update_channels(stub, rebalance.last_hop_pubkey, successful_out) does not take into account MPP

### ISSUE THREAD

**Issue by BhaagBoseDK** [cryptosharks131/lndg#98]
Few issues in update_channels after successful rebalance in rebalancer.py

- [ ] `successful_out = payment_response.htlcs[0].route.hops[0].pub_key` is set to pub_key instead of chan_id. As per [grpc specs](https://api.lightning.community/?python#lnrpc-hop) pub_key is optional and in any case we miss the information related to actual channel ID used in the hop.
- [ ] Also it is only passing first htlc attempt, if the rebalanced was routed over MPP, it is missing out on other attempts.
- [ ] Is update_channels needed in any case? since jobs.py would be updating the channel balances in parallel anyway?

**Comment by cryptosharks131** [comment]
The `update_channels` here is used to update channel balances to prevent rebalancing past your set points while jobs.py is waiting to make its next run, which can be 15-20 seconds later.

**Comment by BhaagBoseDK** [comment]
got it, 
so safer to update all channels for last hop pub key for incoming 
and 
update all outgoing channels based on MPP first hops.

**Comment by cryptosharks131** [comment]
Yeah that is right.

Also agree that using the chan_id is probably the better solution for successful_out (in addition to checking all MPP channels). I think pubkey was used initially because it helps to filter the [ListChannels gRPC](https://api.lightning.community/?python#listchannels) request to a single peer. Using chan_id we would just need to get the pubkey ourselves from our database before making the gRPC call.

We could probably change the update_channels to pass a list of outgoing_channel(s) instead of just a single object, in order to address the MPP issue.

**Comment by BhaagBoseDK** [comment]
A suggestion: 
Considering jobs.py runs regularly updating all channels, and purpose of this update is to ensure before the next rebalance is fired, latest balances should be available, why not just update all channels here as well? It would be be as good as running jobs one additional time.

**Comment by cryptosharks131** [comment]
Updating all channels can mean increasing the time between attempts, especially for nodes who have a lot of channels to update.

Increasing the time frequency of jobs.py might make the site data seem stale at times, as the delay will then be up to 60 seconds instead of 15 seconds. The faster the cycle this runs on, the more the site will feel real-time.

**Comment by BhaagBoseDK** [comment]
then another solution could be to send `payment_response.htlcs[]` to `update_channels()` and in that function, it can iterate over the routes for the first and last hop channel and update the balances for those channels.

**Comment by cryptosharks131** [comment]
Agreed, we should only need to build a list of pubkeys for the outgoing channels in this instance, preferably making sure the list is also unique.

Updating `successful_out` to be a list of outgoing pubkeys should suffice for this, along with updating the function to handle a list of pubkeys.

Another issue with the current `update_channels` function is that it is only updating the first channel returned by `ListChannels` instead of every channel returned in the list.
